libraries/DataFlash/LogFile.cpp

```diff
@@ -1394,66 +1394,47 @@ void DataFlash_Class::Log_Write_AttitudeView(AP_AHRS_View &ahrs, const Vector3f
     WriteBlock(&pkt, sizeof(pkt));
 }
 
-void DataFlash_Class::Log_Write_Current_instance(const uint64_t time_us,
-                                                 const uint8_t battery_instance,
-                                                 const enum LogMessages type,
-                                                 const enum LogMessages celltype)
+void DataFlash_Class::Log_Write_Battery()
 {
-    AP_BattMonitor &battery = AP::battery();
-    float temp;
-    bool has_temp = battery.get_temperature(temp, battery_instance);
-    struct log_Current pkt = {
-        LOG_PACKET_HEADER_INIT(type),
-        time_us             : time_us,
-        voltage             : battery.voltage(battery_instance),
-        voltage_resting     : battery.voltage_resting_estimate(battery_instance),
-        current_amps        : battery.current_amps(battery_instance),
-        current_total       : battery.consumed_mah(battery_instance),
-        consumed_wh         : battery.consumed_wh(battery_instance),
-        temperature         : (int16_t)(has_temp ? (temp * 100) : 0),
-        resistance          : battery.get_resistance(battery_instance)
-    };
-    WriteBlock(&pkt, sizeof(pkt));
+    const uint64_t time_us = AP_HAL::micros64();
+    const AP_BattMonitor &battery = AP::battery();
+    const uint8_t num_instances = battery.num_instances();
 
-    // individual cell voltages
-    if (battery.has_cell_voltages(battery_instance)) {
-        const AP_BattMonitor::cells &cells = battery.get_cell_voltages(battery_instance);
-        struct log_Current_Cells cell_pkt = {
-            LOG_PACKET_HEADER_INIT(celltype),
+    for (uint8_t i = 0; i < num_instances; i++) {
+        float temp;
+        bool has_temp = battery.get_temperature(temp, i);
+        struct log_Battery pkt = {
+            LOG_PACKET_HEADER_INIT(LOG_BATTERY_MSG),
             time_us             : time_us,
-            voltage             : battery.voltage(battery_instance)
+            instance            : i,
+            voltage             : battery.voltage(i),
+            voltage_resting     : battery.voltage_resting_estimate(i),
+            current_amps        : battery.current_amps(i),
+            current_total       : battery.consumed_mah(i),
+            consumed_wh         : battery.consumed_wh(i),
+            temperature         : (int16_t)(has_temp ? (temp * 100) : 0),
```

Comment:
Maybe int16_t-min or int16_t-max?

Reply:
This was prompted by users not liking their logs showing a large value before the battery was detected. (Same way they wanted the per cell voltage to not be a large value before connected). It's a cave in to the users over sanity for log tools.